src/Files.App/Views/Shells/BaseShellPage.cs

```diff
@@ -570,17 +570,31 @@ public virtual void Back_Click()
 			var previousPageContent = ItemDisplay.BackStack[ItemDisplay.BackStack.Count - 1];
 			HandleBackForwardRequest(previousPageContent);
 
-			if (ItemDisplay.CanGoBack)
-				ItemDisplay.GoBack();
+			try
+			{
+				if (ItemDisplay.CanGoBack)
+					ItemDisplay.GoBack();
+			}
+			catch (Exception ex)
+			{
+				App.Logger.LogWarning(ex, "Failed to navigate");
```

Comment:
Could you add a comment for this like below?
```suggestion
				// NOTE: #16104 Trying to locate the origin of crashes, should be removed later.
				App.Logger.LogWarning(ex, "Failed to navigate");
```

Also, others below.

Reply:
Ah this is not for the main branch yet